### PR TITLE
Fix artefact on line junctions when calling `arc()`

### LINKED ISSUE #488: Improve arc rendering
There are currently two methods to render arcs with `fpdf2`:
* [arc()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.arc)
* [solid_arc()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.solid_arc)

However artefacts are visibles on line junctions when generating those shapes with `start_from_center=True` & `end_at_center=True`:

image

Whereas it should be like this (image from HexaPDF):

image

The solution here is probably to use the `h` path operator, intead of closing arcs with a line using `l`:
_cf._ [fpdf/fpdf.py#L1651](https://github.com/PyFPDF/fpdf2/blob/2.5.5/fpdf/fpdf.py#L1651)

You will find all required information about path operators in the [1.7 PDF spec](https://christianhaider.de/dokuwiki/lib/exe/fetch.php?media=pdf:pdf32000_2008.pdf).
Reading the existing code of the `arc()` & `solid_arc()` methods should also help you understand how shapes are drawn.

This is also an opportunity to add unit tests combining `start_from_center=True` & `end_at_center=True` in [test/shapes/test_arc.py](https://github.com/PyFPDF/fpdf2/blob/master/test/shapes/test_arc.py).
Please read the related documentation to get information on how to add tests: https://pyfpdf.github.io/fpdf2/Development.html#testing
I suggest the following approach: taking inpiration from the existing test functions,
write a basic usage scenario that calls `assert_pdf_equal(pdf, "new_test_name.pdf", tmp_path, generate=True)`, and run `pytest -vv test/shape/test_arc.py -k new_test_name`
The `generate=True` argument is very useful to write new tests as it will generate the reference PDF file.
Once you'll be happy with the PDF file generated, you will just have to remove `generate=True` and you'll get a fine non-regression unit test.

Check #258 for more context

Pleae feel free to ask any question regarding this task or `fpdf2` 😊

---

By implementing this feature you, as a benevolent [FLOSS](https://en.wikipedia.org/wiki/Free/libre_and_open-source_software) developper, will improve a useful feature of this library, benefiting the large community of fpdf2 users.
As a contributor you will get review feedbacks from maintainers & other contributors, and learn about the lifecycle & structure of a Python library on the way.
You will also be added to the [contributors list & map](https://github.com/PyFPDF/fpdf2#contributors-).

### PR DESCRIPTION
**Pull request by Jmillan-Dev**
Fixes #488

By using the operator `h` when both `start_from_center` and `end_at_center` are **True** fixes the rendering issue.

This closes the path of the shape instead of leaving it open like `l` did before.

Tests have been added to check this new behavior.

**Checklist**:

- [ ] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [ ] This PR is ready to be merged 

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ in order to get warned about deprecated features used in your code.
 This can also be enabled programmatically with `warnings.simplefilter('default', DeprecationWarning)`.
 
 ## [2.5.7] - not released yet
+### Fixed
+- `arc()` not longer renders artefacts at intersection point [#488](https://github.com/PyFPDF/fpdf2/issues/488)
```
Comment (Lucas-C) [inline CHANGELOG.md:21]: You can add "- thanks to @Jmillan-Dev" if you want!

File: fpdf/fpdf.py [hunk 2]
```diff
@@ -1672,9 +1672,12 @@ def derivative_evaluate(eta):
             )
 
         if end_at_center:
-            self._out(
-                f"{cx * self.k:.2f} {(self.h - cy) * self.k:.2f} l {style.operator}"
-            )
+            if start_from_center:
+                self._out(f"h {style.operator}")
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1676]: 👍

**Comment by Lucas-C** [comment]
@all-contributors please add @Jmillan-Dev for code

**Review by Lucas-C — APPROVED** [review]
Looks great!
Thank you for contributing 😊

**Comment by Lucas-C** [comment]
Seems like some unit tests are now failing:
```
FAILED test/shapes/test_arc.py::test_arc_start_from_center_and_end_at_center
FAILED test/shapes/test_solid_arc.py::test_solid_arc_not_circle
FAILED test/shapes/test_solid_arc.py::test_solid_arc_style
FAILED test/shapes/test_solid_arc.py::test_solid_arc_line_width
FAILED test/shapes/test_solid_arc.py::test_solid_arc_draw_color
FAILED test/shapes/test_solid_arc.py::test_solid_arc_fill_color
FAILED test/shapes/test_solid_arc.py::test_solid_arc_inclination
FAILED test/shapes/test_solid_arc.py::test_arc_clockwise
```

Can you reproduce this on your computer with `pytest -vv test/shapes/test_solid_arc.py`?

I you check the source code of the `solid_arc()` method you'll see that it always calls `arc()` with `start_from_center=True` and `end_at_center=True`. Hence you will have to regenerate all reference PDF files of test function in `test/shapes/test_solid_arc.py` by passing `generate=True` once to all calls to `assert_pdf_equal()`, and then include the newly generated PDF files to this PR.

**Comment by Jmillan-Dev** [comment]
Okay, I forget to change the new test PDF file for `test_arc_start_from_center_and_end_at_center` when I did a last minute change when I read that the operator `h` did not required any arguments.

And I now have updated all of the `solid_arc()` test files since they use `arc(start_from_center=True, end_at_center=True)`

**Comment by Lucas-C** [comment]
This looks great! Merging now

**Pull request merged**